ls1intum/Artemis
# `Discussion`: Replace paginator of discussion page with infinite scrolling

### Checklist
#### General



#### Client

### Motivation and Context



### Description


### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with more than 50 posts 

1. Log in to Artemis
2. Navigate to Course Discussion page
3. Scroll to the bottom of the page and verify that more posts are fetched from the server by focusing on the location of the scroll bar.
4. Check that all filters, sorting, pinning, and achieving of posts still work correctly.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-discussion.component.ts | 82.02% | 95.83% |
| metis.service.ts | 78.57% | 93.95% |



### Screenshots

https://user-images.githubusercontent.com/67344158/160439521-a4aba8bd-bd1f-44a6-8736-e22f9f17885e.mp4
